_posts/jared_baker/2017-03-24-Migrating-Instances.md

```diff
@@ -0,0 +1,91 @@
+
+Here in the Collaboratory we do make use of the migration functionality but in a case by case basis and not typically for instances running CPU intensive workloads.
+
+#### Official documentation
+* [Openstack admin guide on migrations](https://docs.openstack.org/admin-guide/cli-nova-migrate.html)
+* [Openstack admin guide on live migration](https://docs.openstack.org/admin-guide/compute-live-migration-usage.html)
+* [Openstack configuring guide on migrations](https://docs.openstack.org/admin-guide/compute-configuring-migrations.html)
+* [KVM Migration](https://www.linux-kvm.org/page/Migration)
+
+#### Prerequisites
+* [Enable SSH between compute nodes](https://docs.openstack.org/admin-guide/cli-nova-migrate-cfg-ssh.html#clinovamigratecfgssh)
+
+## Migration types
+* Cold Migration
+  * Requires downtime and shared storage. No way to specify destination hypervisor. Very fast.
+* Live Migration
+  * No downtime, requires shared storage. Very fast.
+* Cold Block Migration
+  * Requires downtime, does not require shared storage. Uses local disk on the host to store & run the instances. May take a long time depending on the amount of data that needs to be migrated. No way to specify destination hypervisor.
+* Live Block Migration
+  * No downtime and no requirement for shared stored. If you run instances on local disk and need to migrate without interruptions, this is the method you need. May take a long time depending on the amount of data that needs to be migrated and how active (IO & memory) the instance is.
+
+## How to migrate
+
+Migrating instances can be done from the openstack cli or from the dashboard.
+
+~~~bash
+# Cold Migration
+nova migrate <instance id>
+# Block migration
+nova migrate --block-migrate <instance id>
+# Live migration
+nova live-migration <instance id> <optional: destination hypervisor>
+# Live block migration
+nova live-migration --block-migrate <instance id> <optional: destination hypervisor>
+~~~
+
+Cold migrations don't allow you to specify a destination hypervisor. Instead, the nova scheduler will look for available resources in the cluster and place them accordingly.
+
+#### Migration using the Nova client
+<video width="100%" height="auto" controls preload>
+    <source src="{{site.urlimg}}jared_baker/migratinginstances/migrate-with-ping.webm"></source>
+</video>
+
+#### Migration using the Openstack Dashboard
+<video width="100%" height="auto" controls preload>
+    <source src="{{site.urlimg}}jared_baker/migratinginstances/gui-migrate.webm"></source>
+</video>
+
+## Limitations / Caveats
+* Migrations can take longer or even fail if the rate of change to the disk and/or memory contents outpaces the network transfer rate. With a 10Gbps network this should allow lots of headroom in migrating a busy instance.
+* In environments with mixed CPU models you need to set the following flags in your /etc/nova/nova.conf to ensure migration compatibility
```

Comment:
Maybe add a note saying that this change only affects newly provisioned instances.
Also, if you are not using "cpu_mode=host-passthrough" then the CPU performance of the VM might be negatively impacted because  some of the native capabilities (e.g. SSE4_2) will not be presented to the VM's CPU.